src/test/groovy/graphql/schema/idl/SchemaGeneratorTest.groovy

```diff
@@ -697,4 +697,98 @@ class SchemaGeneratorTest extends Specification {
         then:
         schema.getSubscriptionType().name == "Subscription"
     }
+
+
+    def "comments are used as descriptions"() {
+        given:
+        def spec = """
+        # description 1
+        # description 2
+        type Query {
+            # description 3
+            foo: String
+            union: Union
+            interface(input: Input): Interface
+            enum: Enum
+        }
+        # description 4
+        union Union = Query
+        
+        # description 5
+        interface Interface {
+            # interface field
+            foo: String
+        }
+        # description 6 
+        input Input {
+            # input field
+            foo: String
+        }
+        # description 7
+        enum Enum {
+            # enum value
+            FOO
+        }
+        schema {
+          query: Query
+        }
+        """
+        when:
+        def wiring = RuntimeWiring.newRuntimeWiring()
+                .type("Union", buildResolver())
+                .type("Interface", buildResolver())
+                .build()
+        def schema = generateSchema(spec, wiring)
+
+        then:
+        schema.getQueryType().description == "description 1\ndescription 2"
+        schema.getQueryType().getFieldDefinition("foo").description == "description 3"
+        ((GraphQLUnionType) schema.getType("Union")).description == "description 4"
+
+        ((GraphQLInterfaceType) schema.getType("Interface")).description == "description 5"
+        ((GraphQLInterfaceType) schema.getType("Interface")).getFieldDefinition("foo").description == "interface field"
+
+        ((GraphQLInputObjectType) schema.getType("Input")).description == "description 6"
+        ((GraphQLInputObjectType) schema.getType("Input")).getFieldDefinition("foo").description == "input field"
+
+        ((GraphQLEnumType) schema.getType("Enum")).description == "description 7"
+        ((GraphQLEnumType) schema.getType("Enum")).getValue("FOO").description == "enum value"
+    }
+
+    def "comments are separated from descriptions with empty lines"() {
+        given:
+        def spec = """
+        # should be ignored comment
+        #
+        # description 1
+        # description 2
+        type Query {
+            # this should be ignored
+            # and this
+            #
+            # and this after an empty line
+            # and the last one that should be ignored
+            # 
+            # description 3
+            # description 4
```

Comment:
If we say we dont trim (and I dont think we should) then we should have a description that has preceding and trailing whitespace

eg:

    #      \t\rdescription 4     \t\r